changes/2581-MaratBR.md

```diff
@@ -0,0 +1,2 @@
+Fixes [#2581](/samuelcolvin/pydantic/issues/2581). Using constraints such as `gt` or `le` 
```

Comment:
```suggestion
Allow constraints such as `gt` or `le` on generic models.
```

You could also update the file name to use your handle.

Reply:
I wanted to give MaratBR credit as it was very insightful, I would not have fixed this without that PR for reference (even if mostly because I don't want to be thinking about v1).